.github/workflows/test-deploy-main.yml

```diff
@@ -11,300 +11,22 @@ on:
-
-      - name: Build ${{ matrix.example }} example
-        run: yarn workspace ${{ matrix.example }}-example build
-
-      - name: Start ${{ matrix.example }} example
-        run: yarn workspace ${{ matrix.example }}-example start & npx wait-on -c waitOnConfig.json -t 20000 http-get://localhost:3000/ui/components/authenticator/sign-in-with-username
-
-      - name: Run E2E tests against ${{ matrix.example }} example
-        run: yarn workspace e2e test:examples
-        env:
-          # Override on the default value in `cypress.json` with framework-specific tag
-          TAGS: '${{ matrix.tags }}'
-
-          # Env values for testing flows
-          DOMAIN: ${{ secrets.DOMAIN }}
-          PHONE_NUMBER: ${{ secrets.PHONE_NUMBER }}
-          USERNAME: ${{ secrets.USERNAME }}
-          NEW_PASSWORD: ${{ secrets.NEW_PASSWORD }}
-          VALID_PASSWORD: ${{ secrets.VALID_PASSWORD }}
-  docs:
-    # Only run docs tests if e2e tests pass
-    needs: [setup, unit]
-    runs-on: ubuntu-latest
-    environment: ci
-    env:
-      NODE_ENV: test
-    steps:
-      - name: Checkout Amplify UI
-        uses: actions/checkout@v2
-        with:
-          persist-credentials: false
-
-      - name: Setup Node.js LTS
-        uses: actions/setup-node@v2
-        with:
-          node-version: lts/*
-          cache: 'yarn'
-
-      - name: Restore cypress runner Cache
-        uses: actions/cache@v2
-        id: restore-cypress-cache
-        with:
-          path: ~/.cache/Cypress
-          key: ${{ runner.os }}-cypress-${{ hashFiles('yarn.lock') }}
-
-      - name: Install packages
-        run: yarn install
-
-      - name: Build ui package
-        run: yarn ui build
-
-      - name: Build react package
-        run: yarn react build
-
-      - name: Build docs package
-        run: yarn docs build
-        env:
-          SITE_URL: ${{ secrets.SITE_URL }}
-
-      - name: Start docs site
-        run: yarn docs start & npx wait-on -c waitOnConfig.json -t 20000 http-get://localhost:3000
-
-      - name: Run E2E tests against docs
-        run: yarn workspace e2e test:theme
-        env:
-          # Env values for testing flows
-          DOMAIN: ${{ secrets.DOMAIN }}
-          PHONE_NUMBER: ${{ secrets.PHONE_NUMBER }}
-          USERNAME: ${{ secrets.USERNAME }}
-          NEW_PASSWORD: ${{ secrets.NEW_PASSWORD }}
-          VALID_PASSWORD: ${{ secrets.VALID_PASSWORD }}
+  test:
+    uses: aws-amplify/amplify-ui/.github/workflows/reusable-e2e.yml@main
+    with:
+      commit: ${{ github.sha }}
+      repository: ${{ github.repository }}
+    secrets:
+      AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+      AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+      DOMAIN: ${{ secrets.DOMAIN }}
+      PHONE_NUMBER: ${{ secrets.PHONE_NUMBER }}
+      USERNAME: ${{ secrets.USERNAME }}
+      NEW_PASSWORD: ${{ secrets.NEW_PASSWORD }}
+      VALID_PASSWORD: ${{ secrets.VALID_PASSWORD }}    
```

Comment:
Here's what the caller workflow syntax looks like! 
- `uses:` calls the workflow we created above. 
- `with:` passes the `input` fields.
- `secrets:` passes the secrets reusable workflow needs. 

While it seems redundant to pass repo secrets here when the reusable workflow already has access to them, this is a weird syntax limitation by GitHub actions [discussions](https://github.community/t/reusable-workflows-secrets-and-environments/203695/46).